libfreerdp/codec/include/bitmap.c

```diff
@@ -291,7 +291,7 @@ static INLINE BOOL RLEDECOMPRESS(const BYTE* pbSrcBuffer, UINT32 cbSrcBuffer, BY
 					return FALSE;
 				pbSrc = pbSrc + advance;
 
-				if (!buffer_within_range(pbSrc + sizeof(fgPel), pbEnd))
+				if (!buffer_within_range(pbSrc + PIXEL_SIZE, pbEnd))
```

Comment:
I also have investigated that bug, and here `fgPel` is present only in the case of "Set" commands (in my understanding of the specs). So either the check must be moved a few lines below when we test for `LITE_SET_FG_FGBG_IMAGE` and `MEGA_MEGA_SET_FGBG_IMAGE` or we compute an `extra` var depending on if it's a "set" command, and we check for `buffer_within_range(pbSrc + extra, pbEnd)`

Reply:
good point, checked all locations again.